### PR TITLE
[FLINK-4573] Fix potential resource leak due to unclosed RandomAccess…

### PR DESCRIPTION
**Pull request by lw-lin**
…File in TaskManagerLogHandler

--

Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/TaskManagerLogHandler.java [hunk 1]
```diff
@@ -210,7 +210,17 @@ public void onSuccess(Object filePathOption) throws Throwable {
...
+						long fileLength;
+						try {
+							fileLength = raf.length();
+						} catch (IOException ioe) {
+							display(ctx, request, "Displaying TaskManager log failed.");
+							LOG.error("Displaying TaskManager log failed.", ioe);
+							if (raf != null) {
+								raf.close();
+							}
+							return;
```
Comment (zentol) [inline flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/TaskManagerLogHandler.java]: let's just re-throw the exception to retain current behavior.

File: flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/TaskManagerLogHandler.java [hunk 2]
```diff
@@ -210,7 +210,17 @@ public void onSuccess(Object filePathOption) throws Throwable {
...
 							return;
 						}
-						long fileLength = raf.length();
+						long fileLength;
+						try {
+							fileLength = raf.length();
+						} catch (IOException ioe) {
+							display(ctx, request, "Displaying TaskManager log failed.");
+							LOG.error("Displaying TaskManager log failed.", ioe);
+							if (raf != null) {
```
Comment (lw-lin) [inline flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/TaskManagerLogHandler.java]: @zentol comments addressed; thanks!

**Comment by StephanEwen** [comment]
I think this is good to merge...

**Comment by StephanEwen** [comment]
Merging this...

**Pull request closed without merging**